ontoma/interface.py

```diff
@@ -1,644 +1,224 @@
-# -*- coding: utf-8 -*-
-
-'''main interface class
-'''
+"""Main interface class."""
 
 __all__ = [
-    "OnToma",
-    "make_uri"
-    ]
+    'OnToma',
+]
 
+from collections import namedtuple
 import logging
-import re
-from functools import lru_cache
+import os
+import tempfile
 
-import obonet
+import pandas as pd
 
-from ontoma.downloaders import get_omim_to_efo_mappings, get_ot_zooma_to_efo_mappings
-from ontoma.ols import OlsClient
-from ontoma.zooma import ZoomaClient
+from ontoma import ontology
+from ontoma import owl
+from ontoma.constants import URLS, RESULT_FIELDS
+from ontoma.downloaders import get_manual_xrefs, get_manual_string_mappings
 from ontoma.oxo import OxoClient
-from ontoma.constants import URLS, OT_TOP_NODES
-
+from ontoma.zooma import ZoomaClient
 
 logger = logging.getLogger(__name__)
 
 
+OnTomaResult = namedtuple('OnTomaResult', RESULT_FIELDS)
```

Comment:
I need to return to the user a collection of fields with certain values, and so I was deliberating between a dictionary and a named tuple. In the end I chose a named tuple for two reasons:
1. Using its results looks a bit more neat, for example `ontoma_result.id_full_url` vs. `ontoma_result['id_full_uri']`.
2. More importantly, in the future we might want to add more fields and make the structure of the results more sophisticated; and at some point we might want to convert this to a proper class with some getter methods. This transition could be made much easier if we have a named tuple to start with, rather than a dict

Reply:
Thanks, these are really good point. It was mostly my ignorance that I didn't know about the advantages of named tuples.